### PR TITLE
 perf(pathfinding): rewrite A* with flat grid, tile cache, and zero dependency on Tile/Creature

### PR DESCRIPTION
**Pull request by ramon-bernardo**
### Pull Request Prelude
- [x] I have followed [proper The Forgotten Server code styling][code].
- [x] I have read and understood the [contribution guidelines][cont] before making this PR.
- [x] I am aware that this PR may be closed if the above-mentioned criteria are not fulfilled.

### Changes Proposed
The original A* pathfinding (`AStarNodes` in `map.cpp`) relied on `std::unordered_map`,
`std::unordered_set`, and `std::priority_queue` with dynamic allocation. Profiling showed
~727k `getTile` calls and ~360k hash lookups per stuck creature ("blind rat" scenario).

#### What changed

| File | Change |
|------|--------|
| `src/pathfinding.h` | **New** — `PathFinder` class, `IPathMap` interface, `FindPathParams`, `FrozenPathingConditionCall` |
| `src/pathfinding.cpp` | **New** — A* grid, open list, path reconstruction, condition logic |
| `src/tests/test_pathfinding.cpp` | **New** — 19 unit tests (Boost.Test) |
| `src/map.h` | Removed `AStarNodes`, `AStarNode`, `hashCoord`; added `#include "pathfinding.h"` |
| `src/map.cpp` | Removed all AStarNodes code; `getPathMatching` delegates to `PathFinder` via adapter |
| `src/creature.h` | Moved `FindPathParams`/`FrozenPathingConditionCall` to `pathfinding.h` |
| `src/creature.cpp` | Removed `FrozenPathingConditionCall` implementation |
| `src/CMakeLists.txt` | Added `pathfinding.cpp` |
| `src/tests/CMakeLists.txt` | Added `test_pathfinding` |

#### Performance

- **Flat grid** `Cell[23][23]` on stack → O(1) lookup, zero allocation, no hashing
- **Tile cost cache** in each cell → `getTile` called only on first evaluation (~45k/call), cached thereafter
- **Linear open list** scan O(121) → no `priority_queue` heap operations
- **Single `cellAt()` lookup** per neighbor → replaced separate `getNodeByPosition` + `getCell`
- **Pre-computed pruning flags** → sight-clear conditions evaluated once per `solve()` call
- **Inlined `walkCost`** → `(dx!=0 && dy!=0) ? 25 : 10` instead of function + `std::abs`
- **No `Tile`/`Creature` dependency** → `IPathMap::getWalkCost(x, y) → uint16_t` keeps the algorithm pure

#### Test coverage (19 tests, all passing)

`same_position`, `unreachable`, `single_cardinal_step`, `single_diagonal_step`,
`linear_path_three_steps`, `cost_avoidance`, `zero_cost_blocked`,
`minimal_cost_shortest_path`, `max_search_dist_limits`, `keep_distance`,
`frozen_condition_is_in_range`, `ipathmap_custom_costs`, `ipathmap_all_blocked_except_one`,
`long_path_within_limits`, `constructor_creates_start_node`, `solve_same_position_false_condition`

[cont]: https://github.com/otland/forgottenserver/wiki/Contributing
[code]: https://github.com/otland/forgottenserver/wiki/TFS-Coding-Style-Guide

## Summary by CodeRabbit

* **New Features**
  * Improved pathfinding: sight-aware routing, configurable search/targeting parameters, and more accurate path selection.

* **Refactor**
  * Restructured pathfinding and map sight/line-of-sight logic for cleaner behavior and integration.

* **Tests**
  * Added comprehensive unit tests covering pathfinding scenarios and edge cases.

### CODE REVIEW

File: src/pathfinding.cpp [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+	}
+
+	return bestNode;
+}
+
+uint16_t PathFinder::heuristic(uint16_t x, uint16_t y, uint16_t targetX, uint16_t targetY)
+{
+	const auto differenceX = static_cast<int32_t>(x) - static_cast<int32_t>(targetX);
+	const auto differenceY = static_cast<int32_t>(y) - static_cast<int32_t>(targetY);
+	return static_cast<uint16_t>(differenceX * differenceX + differenceY * differenceY);
```
Comment (Codinablack) [inline src/pathfinding.cpp:176]: I strongly recommend not using the manhattan... which in this case is really just the squared euclidean heuristic for OT grid based pathfinding, since the cost of diagonal is 25 and cardinal is 10, this heuristic is over working to fall short on it's estimation potential anyways... since we don't need multiplication or decimal level precision for this, honestly the best kind of heuristic here is going to be something like this:

` auto h = dx + dy;`
